mydemoproject/demoapp1/models/borrowed_book.py

```diff
@@ -17,23 +18,22 @@ class BorrowedBook(BaseModel):
     user = models.ForeignKey(CustomUser, on_delete=models.CASCADE,
                              related_name='borrowed_books')
     book = models.ForeignKey(Book, on_delete=models.CASCADE)
-    borrow_date = models.DateField(auto_now_add=True)
-    return_date = models.DateField(null=True)
+    borrow_dttm = models.DateTimeField(default=timezone.now)
+    return_dttm = models.DateTimeField(null=True, blank=True)
 
     def __str__(self):
         return f"{self.user.username} borrowed {self.book.book_name}"
 
     @classmethod
-    def can_borrow_book(cls, user, book):
-        if cls.get_all().filter(user=user, book=book,
-                                return_date__isnull=True).exists():
-            return False, ("You already have borrowed this book"
-                           "previously and haven't returned it yet.")
+    def can_borrow_book(cls, user_id, book_id):
+        if cls.objects.filter(user_id=user_id, book_id=book_id,
+                              return_dttm__isnull=True).exists():
+            return False, "Can't borrow book, Return previous one."
+
         return True, "Book can be borrowed"
 
     @classmethod
     def add_borrowed_book(cls, user, book):
-
         available, message = Inventory.available_count(book)
         if not available:
             return False, message
```

Comment:
I think this check can also be moved to the `can_borrow_book` method.

Reply:
done